haas/cli.py

```diff
@@ -79,17 +79,38 @@ def object_url(*args):
         url += '/' + urllib.quote(arg,'')
     return url
 
+def do_request(fn, url, data={}):
+    """Helper function for making HTTP requests against the API.
+
+    Arguments:
+
+        `fn` - a function from the requests library, one of requests.put,
+               requests.get...
+        `url` - The url to make the request to
+        `data` - the body of the request.
+
+    If the environment variables HAAS_USERNAME and HAAS_PASSWORD are
+    defined, The request will use HTTP basic auth to authenticate, with
+    the given username and password.
+    """
+    kwargs = {}
+    username = os.getenv('HAAS_USERNAME')
+    password = os.getenv('HAAS_PASSWORD')
+    if username is not None and password is not None:
```

Comment:
We don't really have any cli tests to speak of, which has been an outstanding issue for a *long* time, as I'm sure you're aware. We have tests for the general schema validation, and the API call's schema doesn't allow an absent or non-string password field. There's not a test for that exact field/call combination, but I'm pretty comfortable with the choke point that the schema library provides; this isn't a class of bugs that we've had problems with, and there's a reason for that. The trouble with testing is there are always missing cases, but I'm not too worried.